### ISSUE TITLE
Aliases for @Get Routes

### ISSUE THREAD

**Issue by ryanpager** [Romakita/ts-express-decorators#255]
## Informations

Type | Version
---|---
Question | 2.x

****

## Description
Is there a way to alias a route in @Get. Im writing a site right now that requires a lot of aliases for the same route (i.e. /account/support, /support, /<lang>/support) -- and its a pain to have to write separate functions for every single route (lots of duplicated code).

## Example

```javascript
// would love a usage something similar to this
@Get('/account/login', '/:language(de|en|es|fr|hu|id|jp|ru)/account/login')
  @SelfReplicated()
  @WithTranslation()
  @Render('login')
  public login(@Request() request: SelfReplicatedRequest, @Response() response: Express.Response): any {
    return Object.assign({}, request.params, {
      page: { title: 'Log In' },
    });
  }

// instead of duplicating methods

@Get('/account/login')
  @SelfReplicated()
  @WithTranslation()
  @Render('login')
  public login(@Request() request: SelfReplicatedRequest, @Response() response: Express.Response): any {
    return Object.assign({}, request.params, {
      page: { title: 'Log In' },
    });
  }

@Get('/:language(de|en|es|fr|hu|id|jp|ru)/account/login')
  @SelfReplicated()
  @WithTranslation()
  @Render('login')
  public loginTranslated(@Request() request: SelfReplicatedRequest, @Response() response: Express.Response): any {
    return Object.assign({}, request.params, {
      page: { title: 'Log In' },
    });
  }
```

**Comment by Romakita** [comment]
Hello @ryanpager ,

For you example this solution works fine:
```typescript
  @Get('/:language(de|en|es|fr|hu|id|jp|ru)?/account/login')
  @SelfReplicated()
  @WithTranslation()
  @Render('login')
  public login(@Request() request: SelfReplicatedRequest, @Response() response: Express.Response): any {
    return Object.assign({}, request.params, {
      page: { title: 'Log In' },
    });
  }
```
> `?` is a optional path. 

Express support regexp with this module https://www.npmjs.com/package/path-to-regexp. You can find some information to write your route expression ;)

See you 
Romain

**Comment by ryanpager** [comment]
Hi @Romakita,

This actually solved some minor issues I was having -- so thanks. There are routes with a ton of aliases which would propagate into crazy endpoints though.

i.e. this is from the old elixir version --

```elixir
    get "/recover", AccountController, :recover
    get "/:locale/recover", AccountController, :recover

    get "/account/recover", AccountController, :recover
    get "/:locale/account/recover", AccountController, :recover

    get "/support/recover", AccountController, :recover
    get "/:locale/support/recover", AccountController, :recover

    get "/forgot-password", AccountController, :recover
    get "/:locale/forgot-password", AccountController, :recover

    get "/account/forgot-password", AccountController, :recover
    get "/:locale/account/forgot-password", AccountController, :recover

    get "/support/forgot-password", AccountController, :recover
    get "/:locale/support/forgot-password", AccountController, :recover
```

**Comment by Romakita** [comment]
Hum I can try to add this feature ;) . I think is not tough to implement that.

**Comment by ryanpager** [comment]
Since im on version 3.10.2 -- is there an easy way for me to grab the router and add those manually somewhere (and trigger the correct controller method). There would be a lot of refactoring I think if I were to upgrade to the newest version.

**Comment by Romakita** [comment]
Between 3 and 4, the difference is `import` module but the old import is with ts-express-decorators works too. Normally, v3 and v4 is compatible (99%).
 
In v3 and v4 hasn't a way to do that, Sorry.

**Comment by Romakita** [comment]
https://www.npmjs.com/package/ts-express-decorators in v4 on npm is also available ;)

**Comment by ryanpager** [comment]
Ok, I will try to get it upgraded. Is there a big difference going to v5?

**Comment by Romakita** [comment]
v5 it's an error of the Travis job, and the release can't removed from Github.

**Comment by Romakita** [comment]
PR =>  #256

**Comment by ryanpager** [comment]
Holy crap this is amazing. Thanks so much for doing that -- do you have a donation page or a "buy me a coffee" page that I can send something as thanks -- saved my bacon here.

**Comment by Romakita** [comment]
@ryanpager Yes, here the link https://opencollective.com/tsed ;) Thanks in advance :D

**Comment by Romakita** [comment]
Released :D v4.1.0
